File: src/python/_1539_Kth_Missing_Positive_Number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        l, r = 0, len(arr) - 1

        while l <= r:
            m = l + (r - l) // 2

            if arr[m] - 1 - m < k:
                l = m + 1
            else:
                r = m - 1

        return l + k

File: src/java/_1539_Kth_Missing_Positive_Number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int l = 0, r = arr.length - 1;

        while (l <= r) {
            int m = l + (r - l) / 2;

            if (arr[m] - 1 - m < k)
                l = m + 1;
            else
                r = m - 1;
        }

        return l + k;
    }
}
